services/src/main/java/org/keycloak/services/clienttype/DefaultClientTypeManagerFactory.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.keycloak.services.clienttype;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.client.clienttype.ClientTypeManager;
+import org.keycloak.client.clienttype.ClientTypeManagerFactory;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.representations.idm.ClientTypeRepresentation;
+import org.keycloak.representations.idm.ClientTypesRepresentation;
+import org.keycloak.util.JsonSerialization;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class DefaultClientTypeManagerFactory implements ClientTypeManagerFactory {
```

Comment:
Is it an option to implement `EnvironmentDependentFactory` and make this provider working just if `Client types` feature is enabled? See other providers relying on the "feature" being enabled.

Reply:
See my other comment (on `DefaultClientTypeProvidetFactory`) regarding this.